packages/lesswrong/server/callbacks/votingCallbacks.ts

```diff
@@ -4,44 +4,46 @@ import { voteCallbacks, VoteDocTuple } from '../../lib/voting/vote';
 import { batchUpdateScore } from '../updateScores';
 
+const collectionsThatAffectKarma = ["Posts", "Comments", "Revisions"];
+
+const usersVoteAppliesTo = (newDocument: DbVoteableType, vote: DbVote): string[] => {
+  // do not update if the voting user is the author or one of the co-authors
+  const userIds: string[] = [newDocument.userId].concat(newDocument.coauthorUserIds || []);
+  if (userIds.includes(vote.userId) || !collectionsThatAffectKarma.includes(vote.collectionName)) return [];
+
+  return userIds;
+}
+
 /**
- * @summary Update the karma of the item's owner
- * @param {object} item - The item being operated on
- * @param {object} user - The user doing the operation
- * @param {object} collection - The collection the item belongs to
- * @param {string} operation - The operation being performed
+ * Share karma among users, giving each user `karma/sqrt(numUsers)`
+ * @param userIds 
+ * @param karma 
  */
-const collectionsThatAffectKarma = ["Posts", "Comments", "Revisions"]
-voteCallbacks.castVoteAsync.add(function updateKarma({newDocument, vote}: VoteDocTuple, collection: CollectionBase<DbVoteableType>, user: DbUser) {
-  // only update karma is the operation isn't done by the item's author
-  if (newDocument.userId !== vote.userId && collectionsThatAffectKarma.includes(vote.collectionName)) {
-    void Users.rawUpdateOne({_id: newDocument.userId}, {$inc: {"karma": vote.power}});
-  }
-});
-
-voteCallbacks.cancelAsync.add(function cancelVoteKarma({newDocument, vote}: VoteDocTuple, collection: CollectionBase<DbVoteableType>, user: DbUser) {
-  // only update karma is the operation isn't done by the item's author
-  if (newDocument.userId !== vote.userId && collectionsThatAffectKarma.includes(vote.collectionName)) {
-    void Users.rawUpdateOne({_id: newDocument.userId}, {$inc: {"karma": -vote.power}});
-  }
-});
+const applyKarmaToUsers = (userIds: string[], karma: number): void => {
+  if (userIds.length === 0) return;
 
+  // round "away from zero" to be generous (and to simplify making votes reversible)
+  const roundFunc = karma > 0 ? Math.ceil : Math.floor
+  const appliedKarma = roundFunc(karma / Math.sqrt(userIds.length));
+  void Users.rawUpdateMany({ _id: {$in: userIds} }, {$inc: {"karma": appliedKarma}});
+}
 
-voteCallbacks.castVoteAsync.add(async function incVoteCount ({newDocument, vote}: VoteDocTuple) {
-  const field = vote.voteType + "Count"
+const updateVoteCount = (newDocument: DbVoteableType, vote: DbVote, increment: number): void => {
+  if (usersVoteAppliesTo(newDocument, vote).length === 0) return;
 
-  if (newDocument.userId !== vote.userId) {
-    void Users.rawUpdateOne({_id: vote.userId}, {$inc: {[field]: 1, voteCount: 1}});
-  }
-});
+  const field = vote.voteType + "Count";
+  void Users.rawUpdateOne({_id: vote.userId}, {$inc: {[field]: increment, voteCount: increment}});
+}
 
-voteCallbacks.cancelAsync.add(async function cancelVoteCount ({newDocument, vote}: VoteDocTuple) {
```

Comment:
these functions used to be async but I don't think they needed to be, correct me if I'm wrong